enhancements/installer/bootstrapping-clusters-with-capi-providers.md

```diff
@@ -0,0 +1,655 @@
+installer-provisioned installs as well as the added ability to produce
+cluster-api manifests, which lay the foundation for future OpenShift features.
+
+## Motivation
+
+- OpenShift Alignment with CAPI: CAPI offers numerous potential benefits;
+such as: day-2 infrastructure management, an API for users to edit cluster
+infrastructure, and upstream collaboration. Installer support for CAPI would
+be foundational for adopting these benefits.
+
+- Terraform BSL License Change: due to the restrictive license change of
+Terraform, `openshift-install` needs a framework to replace the primary
+tool it used to provision cluster infrastructure. In addition to the benefits
+listed above, CAPI provides solutions for the biggest gaps left by Terraform:
+a common API for specifying infrastructure across cloud providers and robust
+infrastructure error handling. Furthermore, removing Terraform will greatly
+reduce the surface area for security vulnerabilities. Terraform and its providers
+constitutes over half a million lines of code in the Installer repo and is directly
+attributable for all (13) CVEs that have been fixed in the Installer repo.
+
+- Streamline Installer development: a common pattern for Installer development
+has been to reimplement control-plane features in Terraform that have already been
+delivered upstream. By utilizing CAPI providers for Day-0 provisioning, our development practices will remove this duplication and become more efficient.
+
+### User Stories
+
+- As an existing user of the installer, I want to continue to use the installer binary (e.g. `create cluster`) in the same environments and with existing automation.
+- As an advanced user or cluster administrator, I want to be able to edit the CAPI infrastructure manifests so that I can customize control-plane infrastructure.
+
+### Goals
+
+- To provide a common experience across platforms for users and `openshift-install` developers
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+  - To document for end users differences in infrastructure created for each cloud provider.
+    Example: on AWS we are expecting there to be additional security groups used, rather than 1 per worker and 1 per master, each machine will be a member of multiple security groups which add permissions based on their roles (node, master, load balancer, most machines will be in 2 of these).
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies (e.g. containers).
+- To maintain compatibility for hive, particularly regarding `destroy` functionality
+- To continue delivering `openshift-install` for supported os/architectures: linux/x86, linux/arm, linux/powerpc, linux/s390x, darwin/x86, darwin/arm
+
+### Non-Goals / Future work
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
```

Comment:
If any of the footprint changes have costs associated with them we will want to make sure that gets into documentation. I expect that's not the case, but lets try to spend some time making sure we understand if it does or not.